pygfx/objects/_base.py

```diff
@@ -134,6 +133,15 @@ def __init__(
         position=None,
     ):
         super().__init__()
+        self._parent: weakref.ReferenceType[WorldObject] = None
+        self.children: List[WorldObject] = []
+
+        self.transform = AffineTransform()
+        self.cache = None
+
+        # @almarklein: Can we delete this one? It feels out of place.
+        if position is not None:
+            self.transform.position = position
```

Comment:
We could discuss that. I personally find it convenient to set the position of simple objects via the constructor. Then again, if we go this route then we should probably do use ``**kwargs`` and allow any prop to be set this way ...

Reply:
I think this will cause a lot of confusion from an API point of view. (1) How would one discover which objects can be set? (2) How should we deal with aliased properties (`obj.transform.position` vs `obj.world_transform.position` [currently unlinked])? (3) What would be the expected behavior when an attribute passed to `kwargs` doesn't exist on the object or it's nested objects?

Reply:
1. We could do the same as we do with the material classes: all props of the object can be passed to the constructor.
2. The `position` was a property of the world object. If after this PR it is not anymore, this init indeed feels much less natural. Let's revisit when this PR is converging towards the new situation. We could remove it for now.
3. `AttributeError` (checking code in materials, it just has a kw-arg for each prop (no `**kwargs`).